databases/week2/courseRegistration.sql

```diff
@@ -0,0 +1,45 @@
+DROP DATABASE IF EXISTS courseRegistration;
+CREATE DATABASE IF NOT EXISTS courseRegistration;
+USE courseRegistration;
+
+CREATE TABLE student (
+    id int NOT NULL PRIMARY KEY AUTO_INCREMENT COMMENT 'Primary Key',
+    student_name VARCHAR(255) NOT NULL COMMENT 'student name',
+    email VARCHAR(255) NOT NULL COMMENT 'email id',
+    phone VARCHAR(255) NOT NULL COMMENT 'phone Number',
+    program VARCHAR(255) NOT NULL
+) DEFAULT CHARSET UTF8 COMMENT 'studentTable';
+
+CREATE TABLE course (
+    id int NOT NULL PRIMARY KEY AUTO_INCREMENT COMMENT 'Primary Key',
+    course_name VARCHAR(255) NOT NULL COMMENT 'course name',
+    credits VARCHAR(255) NOT NULL COMMENT 'credits',
+    prerequisite VARCHAR(255) NOT NULL,
+    syllabus VARCHAR(255) NOT NULL
+) DEFAULT CHARSET UTF8 COMMENT 'courseTable';
+
+CREATE TABLE department (
+    id int NOT NULL PRIMARY KEY AUTO_INCREMENT COMMENT 'Primary Key',
+    department_name VARCHAR(255) NOT NULL COMMENT 'department name'
+) DEFAULT CHARSET UTF8 COMMENT 'departmentTable';
+
+CREATE TABLE instructor (
+    id int NOT NULL PRIMARY KEY AUTO_INCREMENT COMMENT 'Primary Key',
+    instructor_name VARCHAR(255) NOT NULL COMMENT 'instructor name',
+    department_id INT NOT NULL COMMENT 'department id',
+    CONSTRAINT fk_department FOREIGN KEY(department_id) REFERENCES department (id) ON DELETE CASCADE
+) DEFAULT CHARSET UTF8 COMMENT 'instructorTable';
+
+CREATE TABLE registration_details (
+    student_id INT NOT NULL,
+    instructor_id INT NOT NULL,
```

Comment:
Can two different students in the same course have different instructors or does each course have a single instructor? If each course has a single instructor then i suggest having a separate table to connect instructors and courses, otherwise it's good as is :)